Modifications to include OAS for API Proxies and Policies traits

fermujica opened the pull request.

fermujica committed:
Configure draft for RAML/OAS proxy configuration

fermujica committed:
Change RAML for RAML/OAS specs

fermujica committed:
Modify traits for OAS snippets

fermujica committed:
Update title in nav bar

evangelina commented on modules/ROOT/pages/proxy-deploy-raml-oas-proxy.adoc:
```diff
@@ -0,0 +1,28 @@
++
+image::proxy-deploy-raml-oas-proxy-85634.png[]
+. Click *Advanced Options*.
+. In *Proxy Version*, select *v2.0 (Improved API specification parser)*.
```
it is not only Improved API specification parser but also native OAS support. Adding @chiragnarangmulesoft to review wording

evangelina replied:
I think what is missing is a note that v1.0 is deprecated in favor of v2.0 and comment the EOL. @chiragnarangmulesoft do you know what's the stance of the EOL here?

chiragnarangmulesoft replied:
Yes, we should call out that v2.0 is recommended for both OAS/RAML specs and native OAS is only supported with 2.0. If v1.0 is used with OAS spec, then it's translated to RAML. @evangelina agree?

evangelina replied:
Agree, @fermujica is this change clear? Let me know if you rather I make the change

evangelina replied:
Hi @fermujica, can we add a note stating:
"v2.0 is the recommended version for OAS/RAML specs. Native OAS is only supported with version 2.0, if v1.0 is used with OAS spec it is translated to RAML"

fermujica replied:
Please check this note: 031b286

chiragnarangmulesoft commented on modules/ROOT/pages/analytics-faq.adoc:
```diff
@@ -30,10 +30,10 @@ If these messages are not present, check the application's log to see that you s
 ----
 
-In previous versions of Anypoint Data Gateway, an issue prevented RAML type proxies from working correctly. Ensure that you have the latest software where this issue has been fixed.
+In previous versions of Anypoint Data Gateway, an issue prevented RAML/OAS type proxies from working correctly. Ensure that you have the latest software where this issue has been fixed.
```
@evangelina @fermujica why are we talking about Data Gateway here? Is it a typo?

evangelina replied:
Yes, this is a type, @fermujica , can we change Anypoint Data Gateway for API Gateway?

fermujica replied:
You are right. This was a typo: 6e591a5

evangelina commented on modules/ROOT/pages/prepare-raml-task.adoc:
```diff
@@ -26,15 +27,21 @@ And then you can apply this trait in each individual operation like this:
 +
 image::snippet-rate-limiting.png[]
-. Select RAML 0.8 or RAML 1.0 to match the API RAML version.
+.. Or OAS 2.0 to match the API OAS version.
```
One important comment here, only the latest versions of the policies will include the OAS snippets. Since the OAS release was postponed to January we can't share now what those versions are, but we can provide a list of the policies that have the snippets and since which version. What do you think is best @fermujica ?

evangelina replied:
@fermujica, can we add a note saying that only since the following versions the policies contain the OAS snippets?

Rate limiting - SLA based | since v1.1.2
OAuth 2.0 access token enforcement using Mule OAuth provider  | since v1.1.3
OpenAM access token enforcement | since v1.1.3
OpenId Connect access token enforcement | since v1.1.3
OpenId Connect access token enforcement | since v1.1.3
Client ID enforcement | since v1.1.3

fermujica replied:
@evangelina Sure. 
Can I assume that if the policies are not in this table, then they don't support OAS snippets?

fermujica replied:
Let me know if this clarification works: fb8803f

fermujica committed:
"API Data Gateway" should be "API Gateway"

fermujica committed:
v2.0 is the recommended proxy version for OAS specs

fermujica committed:
Add supported policies for OAS Snippets

evangelina commented on modules/ROOT/pages/prepare-raml-task.adoc:
```diff
@@ -26,15 +27,35 @@ And then you can apply this trait in each individual operation like this:
++
+image::prepare-raml-task-2b676.png[]
++
+OAS Snippets is not supported by all policies versions. +
```
@fermujica more than supported is actually displayed/shown.

fermujica replied:
👍 
Tell me if this works better: 92768f0

evangelina commented on modules/ROOT/pages/prepare-raml-task.adoc:
```diff
@@ -26,15 +27,35 @@ And then you can apply this trait in each individual operation like this:
+| OpenId Connect access token enforcement | since v1.1.3
+| Client ID enforcement | since v1.1.3
+|===
 . Edit the API to add the code.
```
@FedeAmdan can you review that the versions are correct?

FedeAmdan replied:
@fermujica The versions are correct, but "OpenId Connect access token enforcement" is repeated. The second time should say: "PingFederate access token enforcement "

fermujica replied:
Fixed in 44b797f

fermujica committed:
Reword note about OAS snippets display for policies

fermujica committed:
non-RAML REST APIs should be non-defined specs (could be either RAML or OAS)

fermujica committed:
Advantages and impacts on requests for each type of API proxy should be on their own article

The "Reviewing API Concepts" article ended up being list of benefits and 
impacts on proxy requests. Renaming the article accordingly.

fermujica committed:
Updating error code descriptions for API proxies

fermujica committed:
Adding links to SSL config instructions for Mule 3 and Mule 4 based API proxies

fermujica committed:
Repeated OpenID enforcement policy should be PingFederate enforcement

fermujica commented:
Adding the following changes based on @FedeAmdan comments:

* 44b797f Repeated OpenID enforcement policy should be PingFederate enforcement
* 20f9c4a Adding links to SSL config instructions for Mule 3 and Mule 4 based API proxies
* e19da2a Updating error code descriptions for API proxies
* addc9b8 Advantages and impacts on requests for each type of API proxy should be on their own article
* 64db3c3 non-RAML REST APIs should be non-defined specs (could be either RAML or OAS)

evangelina reviewed (approved):
LGTM
